be/test/olap/schema_change_test.cpp

```diff
@@ -672,7 +672,505 @@ TEST_F(TestColumn, ConvertVarcharToDate) {
     OLAPStatus st = read_row.convert_from(1, read_row.cell_ptr(0), tp, _mem_pool.get());
     ASSERT_EQ(st, OLAP_ERR_INVALID_SCHEMA);
 }
-  
+
+TEST_F(TestColumn, ConvertVarcharToTinyInt) {
+    AddColumn(
+            "VarcharColumn",
+            "VARCHAR",
+            "REPLACE",
+            255,
+            false,
+            true);
+
+    AddColumn(
+            "ConvertColumn",
+            "TINYINT",
+            "REPLACE",
+            1,
+            false,
+            false);
+
+    TabletSchema tablet_schema;
+    InitTablet(&tablet_schema);
+    CreateColumnWriter(tablet_schema);
+
+    RowCursor write_row;
+    write_row.init(tablet_schema);
+
+    RowBlock block(&tablet_schema);
+    RowBlockInfo block_info;
+    block_info.row_num = 10000;
+    block.init(block_info);
+
+    std::string src_str = "127";
+    write_row.set_field_content(0, src_str.data(), _mem_pool.get());
```

Comment:
For a `VARCHAR` type cell, a `Slice` object will be used in `RowCursor`, but here you passed a real char*, it may be a bug.

Reply:
done